stglib/sig/cdf2nc.py

```diff
@@ -313,19 +370,25 @@ def ds_rename_sig(ds, waves=False):
 
 def fix_encoding(ds):
     """ensure we don't set dtypes uint for CF compliance"""
-    if "sample" not in ds.dims:
+    if "units" in ds["time"].encoding:
+        ds["time"].encoding.pop("units")
+
+    tstep = ds["time"][1] - ds["time"][0]  # use time step to select time encoding
+    # if "sample" not in ds.dims:
+    if tstep < np.timedelta64(1, "m"):
         print(
-            "make time encoding to dtype double because no sample dimension, round to milliseconds first"
+            f"make time encoding to dtype double because tstep {tstep} seconds is < 1 minute, round to milliseconds first"
         )
         ds["time"] = ds["time"].dt.round("ms")  # round time to milliseconds first
         ds["time"].encoding["dtype"] = "double"
-        """if ds[var].max() > 2**31 - 1 or ds[var].min() < -(2**31):
-                print(
-                    f"warning {var} may be too big to fit in int32: min {ds[var].min().values}, max {ds[var].max().values} so make double"
-                )
-               ds[var].encoding["dtype"] = "double"
-               """
+
     else:
+        print(
+            f"make time encoding int because tstep {tstep} seconds is >= 1 minute, round time to seconds first"
+        )
+        ds["time"] = ds["time"].dt.round(
+            "s"
+        )  # round time to seconds if time interval >= 1 minute
```

Comment:
Please put comments like this on their own line so black doesn't do this kind of unnecessary line wrapping

Reply:
changed as suggested.